nerfstudio/engine/trainer.py

```diff
@@ -102,11 +102,14 @@ def __init__(self, config: cfg.Config, local_rank: int = 0, world_size: int = 1)
         writer.put_config(name="config", config_dict=dataclasses.asdict(config), step=0)
         profiler.setup_profiler(config.logging)
 
-    def setup(self, test_mode=False):
+    def setup(self, test_mode: str = "val"):
         """Setup the Trainer by calling other setup functions.
 
         Args:
-            test_mode: Whether to setup for testing. Defaults to False.
+            test_mode: ['val', 'test', 'inference']
```

Comment:
The list of options can be removed since it is in the type definition. The description of each is good though.